### DISCUSSION TITLE
Tricky spacing

### DISCUSSION THREAD

**Discussion by Beiri22** [typst/typst#1593] [Q&A] [answered]
Hi there,
I want to print a task with the number of credit points. If enough space, the credit points should be printed on the last row of the task description, but right aligned. When not enough space, then on the next line, right aligned. I tried to just use #h(1fr) as spacer, but whenever this breaks to the next line, it will not align to the right. In latex i used:
```
% Etwas Platz, dann rechts rücken
% Strut erlaubt, dass es auch auf einer leeren neuen Zeile hinter rückt
\quad\strut\nobreakspace\hfill
	
% Eigentliche Punktzahl in der Klammer
\mbox{(\int_to_arabic:n{#1}~Punkte)}
```

Is there anyone who can help?

**Comment by Julian-Wassmann** [comment, marked as answer]
Hi,

I would suggest this solution: [ReproductionURL](https://typst.app/project/r1k7LlF_064tYt3rdMMlsK)
```
#let customBox(
  mainContent,
  supplement,
  padding: 0em,
) = {
  // block(): optional layout
  block(width: 50%, stroke: black, inset: 1em)[
    // optional styling
    #set par(justify: true)
    #set par(linebreaks: "optimized")
  
    #style(styles => {
      // measure supplement width and add padding
      let supplementWidth = measure(supplement, styles).width + padding
  
      mainContent
      // horizontal spacing keeps space free for supplement
      h(supplementWidth)
      // box fills up remaining line (necessary for align to work)
      box(width: 1fr)[
        #align(right)[
          // wrap content in sized box to prevent line breaks
          #box(width: supplementWidth)[#supplement]
        ]
      ]
    })
  ]
}

// fits at the end of the line
#customBox(lorem(38), [*5 CP*])
// still fits, but justified text is shortened
#customBox(lorem(38), [*5 CP*], padding: 1em)
// doesn't fit because of increased padding
#customBox(lorem(38), [*5 CP*], padding: 2em)
// doesn't fit because of larger supplement
#customBox(lorem(38), [*10 CP*], padding: 1em)
```
Does this work for you?

**Comment by Beiri22** [comment]
This is quiet elegant! But in the following example, I like to show you a problem: https://typst.app/project/rlpBh_MdCUk0DkGzcrEnIC
Can you solve it ;-) ?

**Reply by Julian-Wassmann** [reply]
Solution: `#set par(linebreaks: "optimized")`

**Reply by Beiri22** [reply]
That stops from beaking the word to the next line. But it also stops breaking the word, if getting longer, at all... I modified the example. Please have a look

**Reply by Julian-Wassmann** [reply]
Yes, this is the expected behavior of `linebreaks: "optimized"`. It will also happen without the `customBox()` implementation. However, this shouldn't be a problem. As you already noticed, it only happens for long, artificial words which can't be hyphened. For real scenarios, hyphenation shoud work.

**Reply by Beiri22** [reply]
Thank you!
